lib/resque/data_store.rb

```diff
@@ -106,9 +106,17 @@ def push_to_queue(queue,encoded_item)
         end
       end
 
-      # Pop whatever is on queue
-      def pop_from_queue(queue)
-        @redis.lpop(redis_key_for_queue(queue))
+      # Pop whatever is on queue. Returns the queue name and the object if
+      # popped or nil.
+      def pop_from_queue(*queues, interval: 0)
+        if interval.to_i > 0
```

Comment:
This has to be an integer. If it's a float, redis apparently interprets it like another key name so will end up waiting forever.